network/nodekeeper/nodekeeper.go

```diff
@@ -19,50 +19,66 @@ package nodekeeper
 import (
 	"bytes"
 	"sort"
+	"strings"
 	"sync"
 	"time"
 
+	"github.com/insolar/insolar/configuration"
 	"github.com/insolar/insolar/core"
 	"github.com/insolar/insolar/log"
+	"github.com/insolar/insolar/network/consensus"
+	"github.com/insolar/insolar/network/hostnetwork/transport"
 	"github.com/pkg/errors"
 )
 
-// NodeKeeper manages unsync, sync and active lists
-type NodeKeeper interface {
-	// GetID get current node ID
-	GetID() core.RecordRef
-	// GetSelf get active node for the current insolard. Returns nil if the current insolard is not an active node.
-	GetSelf() *core.ActiveNode
-	// GetActiveNode get active node by its reference. Returns nil if node is not found.
-	GetActiveNode(ref core.RecordRef) *core.ActiveNode
-	// GetActiveNodes get active nodes.
-	GetActiveNodes() []*core.ActiveNode
-	// AddActiveNodes add active nodes.
-	AddActiveNodes([]*core.ActiveNode)
-	// SetPulse sets internal PulseNumber to number. Returns true if set was successful, false if number is less
-	// or equal to internal PulseNumber. If set is successful, returns collected unsync list and starts collecting new unsync list
-	SetPulse(number core.PulseNumber) (bool, *UnsyncList)
-	// Sync initiates transferring syncCandidates -> sync, sync -> active.
-	// If number is less than internal PulseNumber then ignore Sync.
-	Sync(syncCandidates []*core.ActiveNode, number core.PulseNumber)
-	// AddUnsync add unsync node to the unsync list. Returns error if current node is not active and cannot participate in consensus
-	AddUnsync(*core.ActiveNode) error
-	// GetUnsyncHolder get unsync list executed in consensus for specific pulse.
-	// 1. If pulse is less than internal NodeKeeper pulse, returns error.
-	// 2. If pulse is equal to internal NodeKeeper pulse, returns unsync list holder for currently executed consensus.
-	// 3. If pulse is more than internal NodeKeeper pulse, blocks till next SetPulse or duration timeout and then acts like in par. 2
-	GetUnsyncHolder(pulse core.PulseNumber, duration time.Duration) (*UnsyncList, error)
+// NewActiveNodeComponent create active node component
+func NewActiveNodeComponent(configuration configuration.Configuration) (core.ActiveNodeComponent, error) {
```

Comment:
provide all configuration if you really need it

Reply:
i need host and node configuration here